### DISCUSSION TITLE
BLE picked up a floor above the RPI4 -- options to limit?

### DISCUSSION THREAD

**Discussion by gmalbert** [mKeRix/room-assistant#531] [Q&A]
I have an RPI4 in my basement running 2.13 and I'm using the ios companion app with BLE. When I'm a floor above my basement, I get readings as if I'm sitting in the basement (3, 4, 2, 3.5, etc). I find it hard to imagine that BLE is strong enough to penetrate floors but that appears to be what's happening.

That said, is this something I can do to stop this behavior? Would adding another RA instance on my first floor in the same spot eliminate this? Limiting the threshold doesn't seem to make sense on the basement because the distance is well within the threshold if I'm a floor above.

**Comment by gbdavidx** [comment]
minRssi, but i havent found it to work

**Reply by gmalbert** [reply]
@gbdavidx My understanding of minRssi is that it only shows up in Bluetooth Classic, but not BLE or at least that's how I read this: https://www.room-assistant.io/integrations/bluetooth-low-energy.html#settings

I'd also think that the RSSI would be strong one floor up anyway, but maybe I'm wrong.

I wonder if I can do a BT classic RSSI limit but scan on BLE? I wonder if the combo will work.

**Reply by mKeRix** [reply]
You can of course set both, but they won't affect each other. The distance used for comparing against `maxDistance` is calculated from the RSSI, hence why there is no extra setting for it. The unfortunate truth is that these multi-floor arrangements are quite hard to track with Bluetooth in room-assistant atm, you need very well-tuned placements and configs which might be hard to achieve. I have a lot of hope that #507 would resolve these issues, as then you could make use of other instance data to determine presence as well.

**Reply by gmalbert** [reply]
Thank you. It's so wild to me that Bluetooth can go through floors as if I'm right in the room. I'm excited to try #507. Very much appreciate the great work. Happy to keep testing!
